frontend/src/lib/components/HgLogView.svelte

```diff
@@ -5,25 +5,24 @@
 
 <script lang="ts">
   import t from '$lib/i18n';
+  import type { Readable } from 'svelte/store';
 
   import FormatDate from './FormatDate.svelte';
   import type { Circle, Path } from './TrainTracks.svelte';
   import TrainTracks from './TrainTracks.svelte';
+  import Loader from './Loader.svelte';
+  import type { ProjectChangesetsQuery } from '$lib/gql/types';
 
-  type LogEntry = {
-    node: string;
-    parents: string[];
-    date: number[];
-    user: string;
-    desc: string;
-  };
+  type LogEntries = NonNullable<ProjectChangesetsQuery['projectByCode']>['changesets'];
+  type LogEntry = NonNullable<LogEntries>[0];
   type ExpandedLogEntry = LogEntry & {
     trimmedLog: string;
     row: number;
     col: number;
   };
 
-  export let json: LogEntry[]; // JSON-format hg log
+  export let logEntryStore: Readable<{changesets: LogEntries | undefined, loading: boolean}>;
+  $: logEntries = $logEntryStore.changesets; // JSON-format hg log
```

Comment:
can we break this tight coupling of the return value of the load function to this prop? Just keep it simple, have a loading prop and a log entries prop, no store in the component.